x-pack/legacy/plugins/spaces/public/views/management/components/advanced_settings_subtitle/advanced_settings_subtitle.test.tsx

```diff
@@ -4,16 +4,27 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 import React from 'react';
-import { shallowWithIntl } from 'test_utils/enzyme_helpers';
+import { mountWithIntl } from 'test_utils/enzyme_helpers';
 import { AdvancedSettingsSubtitle } from './advanced_settings_subtitle';
+import { EuiCallOut } from '@elastic/eui';
 
 describe('AdvancedSettingsSubtitle', () => {
-  it('renders as expected', () => {
+  it('renders as expected', async () => {
     const space = {
       id: 'my-space',
       name: 'My Space',
       disabledFeatures: [],
     };
-    expect(shallowWithIntl(<AdvancedSettingsSubtitle space={space} />)).toMatchSnapshot();
+
+    const wrapper = mountWithIntl(
+      <AdvancedSettingsSubtitle getActiveSpace={() => Promise.resolve(space)} />
+    );
+
+    await Promise.resolve();
```

Comment:
Yeah, I hate having to do this. I think it's somewhat common, but I can add a comment explaining

Reply:
Okay, if it's common then it's fine without comment

Reply:
This is probably a better way to implement it:

```js
await wait(() => {
  wrapper.update();
  expect(wrapper.find(SpaceAvatar)).toHaveLength(3);
})
```

Reply:
By default, it will try to run the callback every 100ms and if it hasn't completed within 1s, first then will it fail. This is of course configurable.